### ISSUE TITLE
What data shape could be mamba's correct input?

### ISSUE THREAD

**Issue by duansanshi** [Atik-Ahamed/TimeMachine#9]
import torch
from mamba_ssm import Mamba

batch, length, dim = 2, 64, 16
x = torch.randn(batch, length, dim).to("cuda")
model = Mamba(
    # This module uses roughly 3 * expand * d_model^2 parameters
    d_model=dim, # Model dimension d_model
    d_state=16,  # SSM state expansion factor
    d_conv=4,    # Local convolution width
    expand=2,    # Block expansion factor
).to("cuda")
y = model(x)
assert y.shape == x.shape
This is the official introduction of mamba-ssm,so i think the input shape should be B L d,while in timemachine,the input data's shape is B M ni
I am confused at this about this,could you please provide some guidance for me.Thank you so much for your time and assistance.

**Comment by Atik-Ahamed** [comment]
Thanks for your interest in our work. You may think about a similar strategy as BLD, where B is batch size, L can be referred to as M, and D is ni. Please refer to our paper's Integrated Quadruple Mambas subsection for more details.

**Comment by duansanshi** [comment]
Ok, thans a lot, i think i understand it！

**Comment by duansanshi** [comment]
Thanks for answering my question！
So in time machine，the input size is BML，and linear projection1 project it into BMn1，and then use the first mamba to process it. It is quite like iTransformer,which consider the univariate whole time series as a token,and this mamba is to deal with inter-variate correlation. I would appreciate your feedback on my thoughts. Do you think they are on the right track?

**Comment by Atik-Ahamed** [comment]
Somewhat yes. However, it also can deal with each point via transposition, enabling learning both global context and local context simultaneously.

**Comment by duansanshi** [comment]
Ok，I think I know better about your amazing work. Thanks again for your explanation！
